pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexOpenRedirectRule.java

```diff
@@ -76,20 +84,58 @@ private void findSafeLiterals(AbstractApexNode<?> node) {
             int index = literal.jjtGetChildIndex();
             if (index == 0) {
                 if (node instanceof ASTVariableDeclaration) {
-                    addVariable(node);
-                } else {
+                    addVariable((ASTVariableDeclaration) node);
+                } else if (node instanceof ASTBinaryExpression) {
                     ASTVariableDeclaration parent = node.getFirstParentOfType(ASTVariableDeclaration.class);
-                    addVariable(parent);
+                    if (parent != null) {
+                        addVariable(parent);
+                    }
+                    ASTAssignmentExpression assignment = node.getFirstParentOfType(ASTAssignmentExpression.class);
+                    if (assignment != null) {
+                        ASTVariableExpression var = assignment.getFirstChildOfType(ASTVariableExpression.class);
+                        if (var != null) {
+                            addVariable(var);
+                        }
+                    }
 
                 }
             }
+        } else {
+            if (node instanceof ASTField) {
+                try {
+                    final Field f = node.getNode().getClass().getDeclaredField("fieldInfo");
```

Comment:
Sadly, the parser does not populate the information needed. getFieldInfo() returns a null.
Apex Jorje compiler will be open source at some point and stuff like that could be fixed but in the meantime it's still not ready. I don't like use reflection either but it's all I have at this time.